pkg/cli/cmd/resourcetype/show/display.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package show
+
+import (
+	"container/list"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/radius-project/radius/pkg/cli/cmd/resourcetype/common"
+)
+
+// HeadingToSchema holds a nested field path and its schema definition.
+type HeadingToSchema struct {
+	// Heading is the path to this field (e.g. ".database.server.name").
+	Heading string
+	// Schema contains the field's metadata, such as type and nested properties.
+	Schema FieldSchema
+}
+
+// FieldSchema represents the schema of a field in a resource type.
+type FieldSchema struct {
+	// Name is the name of the field.
+	Name string
+	// Type is the type of the field (e.g. "string", "object", etc.).
+	Type string
+	// Description provides additional information about the field.
+	Description string
+	// IsRequired indicates if the field is required.
+	IsRequired bool
+	// IsReadOnly indicates if the field is read-only.
+	IsReadOnly bool
+	// Properties contains nested fields if the type is "object".
+	Properties map[string]FieldSchema
+}
+
+func (r *Runner) display(resourceTypeDetails *common.ResourceType) error {
```

Comment:
Please add a function level comment. Also if some of the names can be made more intuitive that might help.. for example "display" what? Same comment for file name (display.go) and other object names `HeadingToSchema`..

Reply:
Is `Heading` the name of the property here?